app/solutions.py

```diff
@@ -0,0 +1,57 @@
+from .solution_service import is_correct
+from .questions import QUESTION_MAPPING
+
+
+class AttemptError(Exception):
+    def __init__(self):
+        self.description = "Solution or question not found."
+
+    @property
+    def json(self):
+        return {
+            "description": self.description,
+        }
+
+
+class AttemptTypeError(Exception):
+    def __init__(self):
+        self.description = "Solution or question type not found."
+
+    @property
+    def json(self):
+        return {
+            "description": self.description,
+        }
+
+
+class Attempt:
```

Comment:
Should this be more like `AttemptResponse`?
Just going off how it's being used here, you might have greater aspirations

Reply:
Yeah I imagine all of these fairly basic objects will end up being built up in the future. I'll leave it as just `Attempt` as I want to represent the concept of an attempt at a question, rather than some kind of HttpResponse

Reply:
Hmm okay - in that case I think it would make more sense if this object had no concept of JSON, and was entirely seperate from any HTTP concerns, and any of the translation from this to a response would happen in a controller or at a higher service layer

Reply:
Obv okay for now